libs/stats/odc/stats/_cli_save_tasks.py

```diff
@@ -170,13 +180,16 @@ def gqa_predicate(ds):
         sys.exit(3)
 
 
-def _parse_products(dc, products, temporal_range):
-        
-    query = {"product": products}
+def _parse_products(dc, products, temporal_range, dataset_filter):
+
+    filter = {}
+    if dataset_filter:
+        filter = json.loads(dataset_filter)
```

Comment:
I think this code could be moved into the save_tasks function - then we don't need to duplicate this code.